src/shared/parameter_v2/scripts/c_config.py

```diff
@@ -0,0 +1,121 @@
+from c_parameter import CParameter
+from typing import List, Set
+
+#######################################################################
+#                              C Config                               #
+#######################################################################
+
+
+class CConfig(object):
+
+    DEFINITION = "typedef struct {name}_s {{{contents}}} {name}_t;\n"
+    FORWARD_DECLERATION = "typedef struct {name}_s {name}_t;\n"
+    MALLOC = "{name}_t* {name}_config = ({name}_t*)malloc(sizeof({name}_t));\n"
+    INITIALIZATION = "{name}_t {name}_init = {{{contents}}};\n"
+    INCLUDED_CONFIG_INITIALIZATION = ".{name} = {name}_config,\n"
+    MEMCPY = "memcpy({name}_config, &{name}_init, sizeof({name}_t));\n"
+    FREE = "free((void*){ptr_to_instance});\n"
+    INCLUDE_CONFIG = "const {name}_t* {name};\n"
+
+    def __init__(self, config_name: str, ptr_to_instance: str):
+        """Initializes a CConfig with the given name. The corresponding
+        generation strings (definition, forward_decleration, malloc, initialization,
+        memcpy, free, include_config) are available from read-only properties.
+
+        :param config_name: The name of the config (i.e the filename of the yaml)
+        :param ptr_to_instance: A string representation of where this config
+            is located (ex: ThunderbotsConfig->FooConfig)
+
+        """
+        self.config_name = config_name
+        self.ptr_to_instance = ptr_to_instance
+
+        self.parameters: Set[CParameter] = set()
+        self.configs: Set[str] = set()
+
+    def add_parameter(self, parameter: CParameter):
+        """Add a parameter to this config. This parameters represent
+        the parameters defined in the yaml file. When generating the Config
+        the properties from these parametesr will be properly formatted
+        into the appropriate properties.
+
+        :param parameter: The CParameter to add to this config
+
+        """
+        self.parameters.add(parameter)
+
+    def get_parameters(self) -> List[CParameter]:
+        """Returns the list of parameters included in this CConfig
+
+        :returns: List of parameters
+
+        """
+        return self.parameters.copy()
+
+    def include_config(self, config: str):
+        """Add a config to this config to generate.
+
+        :param config: The CConfig to add to this CConfig
+
+        """
+        self.configs.add(config)
+
+    @property
+    def definition(self):
```

Comment:
Would something like `generate_definition` be better to keep it consistent with other file names? (in that the other function names are actions and not just a noun)

Reply:
this is a slightly different style, it actually is meant to replace getters/setters in a class like this which simply represents how it will get generated into a file (properties have much more useful usecases than what I used them for here)

this is a distinction from the other getters in the class (for example get_parameters) which doesn't return a string

Reply:
but the bottom line is that they are very similar https://www.programiz.com/python-programming/property